docs/how-to/tour.rst

```diff
@@ -0,0 +1,122 @@
+arguments, and the result is that it throws ``TypeError: '<kwarg>' is an invalid keyword 
+argument for this function``
+
+Batavia needs those arguments explicitly specified in a standard format so that it can
+check for that case and generate the correct error. The below code examples all use this calling
+convention, and you'll be up to your knees in ``BataviaErrors`` if you're not aware of it.
+
+Building Blocks of Batavia
+**************************
+
+This section is a quick reference for the most common code you'll see.
+
+builtins.js
+^^^^^^^^^^^
+
+.. code-block:: javascript
+
+    var builtins = require('./builtins.js')
+    builtins.abs([-1], {})              // equivalent to abs(-1)
+
+This contains all of the native Python builtin functions, like ``str``, ``len``, and ``iter``.
+
+When dealing with Python types, many of the native JavaScript operations have been modified to
+try to use builtins first. For instance, .toString() will often just call the object's __str__ if
+possible. Still, the best practice is to use the builtins and types wherever possible.
+
+types.js
+^^^^^^^^
+
+.. code-block:: javascript
+
+    var types = require('./types.js)
+    var my_dict = new types.dict([], {})
+
+This contains all of the native Python types that have been implemented in Batavia. It also has some helper functions:
+
+* ``types.js2py`` Converts a native JavaScript type to a corresponding Batavia type.
+* ``types.isinstance`` checks to see if the object is a Python instance of the corresponding type.
+* ``types.isbataviainstance`` checks to see if the object is an instance of **any** Batavia type.
+* ``types.Type.type_name`` get the name of the type.
+
+This allows us to avoid ugly things like comparing ``Object.prototype.constructor``. Instead,
+use ``types.isinstance`` or ``types.isbataviainstance``. Secondly, it's important that the inputs to Python
+types are Pythonized themselves where needed. You should not be making a list() of JavaScript arrays, for
+instance. That doesn't make sense! (It may even pass some tests, which is dangerous.)
+
+core/callables.js
+^^^^^^^^^^^^^^^^^
+
+These methods ensure that all Python code is executed using the proper __call__ procedure, which could be
+overriden or decorated by the programmer.
+
+* ``callables.call_function`` Invokes a function using its __call__ method if possible. If not, just call it normally.
+* ``callables.call_method`` Calls a class method using the call_function specification above.
+* ``callables.iter_for_each`` Exhausts an iterable using the call_function specification above.
+
+As a general rule, use the builtin where possible. If no builtin is available, use the appropriate version
+of ``call_function`` instead of calling Python functions and methods directly. An example:
+
+.. code-block:: javascript
+
+    // Avoid this
+    my_thing.__repr__()
+
+    // Better
+    var callables = require('./core/callables.js')
+    callables.call_method(my_thing, '__repr__', [], {})
```

Comment:
This is actually a bit of a special case. `__repr__` is one of the methods that isn't looked up with getattribute() in cpython. I haven't done a lot of spelunking here yet, and the docs are vague on which methods are in this list, but `__repr__` is explicitly listed.

https://docs.python.org/3/reference/datamodel.html#special-method-lookup